# HG changeset patch
# User welpy-cw <h.w.forms@arcor.de>
# Date 1768150404 -3600
# Node ID 3edfe0ce129885fe64c1261cdf780a3ec87d0d7f
# Parent  29a6b6a5491a5235746b9dad5ae922559340f0a3
Bug 2009042 - Add 'Favorites' as destination for 'Move To' and 'File' buttons. r=arschmitz

These buttons should perform the same actions as the 'Move To' menu items in
the folder and thread pane context menus, as well as in the main menu under
'Message.' Additionally, replace all remaining instances of 'Recent' with
'Recent Destinations,' as started in Bug 1930707.

Differential Revision: https://phabricator.services.mozilla.com/D277573

diff --git a/mail/base/content/messageWindow.xhtml b/mail/base/content/messageWindow.xhtml
--- a/mail/base/content/messageWindow.xhtml
+++ b/mail/base/content/messageWindow.xhtml
@@ -254,12 +254,16 @@
                      tooltiptext="&fileButton.tooltip;"
                      oncommand="goDoCommand('cmd_moveMessage', event.target._folder)">
         <menupopup is="folder-menupopup" id="button-filePopup"
                    mode="filing"
+                   showFileHereLabel="true"
                    showRecent="true"
-                   showFileHereLabel="true"
-                   recentLabel="&moveCopyMsgRecentMenu.label;"
-                   recentAccessKey="&moveCopyMsgRecentMenu.accesskey;"/>
+                   recentLabel="&contextMoveCopyMsgRecentDestinationMenu.label;"
+                   recentAccessKey="&contextMoveCopyMsgRecentDestinationMenu.accesskey;"
+                   showFavorites="true"
+                   favoritesLabel="&contextMoveCopyMsgFavoritesMenu.label;"
+                   favoritesAccessKey="&contextMoveCopyMsgFavoritesMenu.accesskey;"/>
+
       </toolbarbutton>
       <toolbarbutton id="button-showconversation"
                      class="toolbarbutton-1"
                      label="&openConversationButton.label;"
diff --git a/mail/base/content/messenger-menubar.inc.xhtml b/mail/base/content/messenger-menubar.inc.xhtml
--- a/mail/base/content/messenger-menubar.inc.xhtml
+++ b/mail/base/content/messenger-menubar.inc.xhtml
@@ -1080,10 +1080,10 @@
           <menupopup is="folder-menupopup"
                       mode="filing"
                       showFileHereLabel="true"
                       showRecent="true"
-                      recentLabel="&moveCopyMsgRecentMenu.label;"
-                      recentAccessKey="&moveCopyMsgRecentMenu.accesskey;"
+                      recentLabel="&contextMoveCopyMsgRecentDestinationMenu.label;"
+                      recentAccessKey="&contextMoveCopyMsgRecentDestinationMenu.accesskey;"
                       showFavorites="true"
                       favoritesLabel="&contextMoveCopyMsgFavoritesMenu.label;"
                       favoritesAccessKey="&contextMoveCopyMsgFavoritesMenu.accesskey;"/>
         </menu>
@@ -1094,10 +1094,10 @@
           <menupopup is="folder-menupopup"
                       mode="filing"
                       showFileHereLabel="true"
                       showRecent="true"
-                      recentLabel="&moveCopyMsgRecentMenu.label;"
-                      recentAccessKey="&moveCopyMsgRecentMenu.accesskey;"
+                      recentLabel="&contextMoveCopyMsgRecentDestinationMenu.label;"
+                      recentAccessKey="&contextMoveCopyMsgRecentDestinationMenu.accesskey;"
                       showFavorites="true"
                       favoritesLabel="&contextMoveCopyMsgFavoritesMenu.label;"
                       favoritesAccessKey="&contextMoveCopyMsgFavoritesMenu.accesskey;"/>
         </menu>
diff --git a/mail/components/unifiedtoolbar/content/unifiedToolbarPopups.inc.xhtml b/mail/components/unifiedtoolbar/content/unifiedToolbarPopups.inc.xhtml
--- a/mail/components/unifiedtoolbar/content/unifiedToolbarPopups.inc.xhtml
+++ b/mail/components/unifiedtoolbar/content/unifiedToolbarPopups.inc.xhtml
@@ -35,12 +35,15 @@
             hidden="true" />
 </menupopup>
 <menupopup is="folder-menupopup" id="toolbarMoveToPopup"
            mode="filing"
+           showFileHereLabel="true"
            showRecent="true"
-           showFileHereLabel="true"
-           recentLabel="&moveCopyMsgRecentMenu.label;"
-           recentAccessKey="&moveCopyMsgRecentMenu.accesskey;"
+           recentLabel="&contextMoveCopyMsgRecentDestinationMenu.label;"
+           recentAccessKey="&contextMoveCopyMsgRecentDestinationMenu.accesskey;"
+           showFavorites="true"
+           favoritesLabel="&contextMoveCopyMsgFavoritesMenu.label;"
+           favoritesAccessKey="&contextMoveCopyMsgFavoritesMenu.accesskey;"
            oncommand="goDoCommand('cmd_moveMessage', event.target._folder);event.stopPropagation()"/>
 <menupopup is="folder-menupopup" id="toolbarFolderLocationPopup"
            class="menulist-menupopup"
            mode="notDeferred"
